src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -85,18 +86,28 @@ module Inotify = struct
       | None -> Error "invalid message (event type missing)")
 end
 
+let special_file_for_inotify_sync () =
+  Path.build (Path.Build.relative Path.Build.root "dune-inotify-sync")
+
 let command ~root =
-  let excludes =
-    [ {|/_build|}
-    ; {|/_opam|}
+  let exclude_patterns =
+    [ {|/_opam|}
     ; {|/_esy|}
     ; {|/\..+|}
     ; {|~$|}
     ; {|/#[^#]*#$|}
     ; {|4913|} (* https://github.com/neovim/neovim/issues/3460 *)
     ]
   in
-  let path = Path.to_string_maybe_quoted root in
+  let exclude_paths =
+    (* these paths are used as patterns for fswatch, so they better not contain
+       any regex-special characters *)
+    [ "_build" ]
```

Comment:
FTR, it's the same story for `_opam` and `_esy`. So it would make sense to move them here as well.

Reply:
I was cautious here because I thought people might want to have the nested `_opam` and `_esy` directories that are not at the root of the tree, but I just found another issue:
if the directory ignored with "@" is deleted and re-created then it's no longer ignored. (same if the directory was not there when watches were initially established)

This means that if the user creates an _opam or _esy directory while dune is running then a pattern is the only way to catch that.

I'm thinking this is not a problem for _build because we're not trying to be resilient to _build being deleted.

Reply:
Make sense